src/Model/Storage/Filesystem.php

```diff
@@ -96,8 +97,7 @@ private function createFoldersForFile(string $file)
 
 		if (!is_dir($path)) {
 			if (!mkdir($path, 0770, true)) {
-				$this->logger->warning('Failed to create dir.', ['path' => $path]);
-				throw new StorageException($this->l10n->t('Filesystem storage failed to create "%s". Check you write permissions.', $path));
+				throw new StorageException(sprintf('Filesystem storage failed to create "%s". Check you write permissions.', $path));
```

Comment:
Why didn't you keep the localized error messages?

Reply:
There's no need for localized errors at backend exceptions. They aren't meant to be printed at the frontend, so it could happen that there's a catch (and a log) and so you would get a (user-based) translated exception message inside your log :D

Reply:
It could happen, but does it happen?

Reply:
OK, but it's one of the rare places where we use translated exceptions .. But I don't see a use case for it .. what's the reason for translated exceptions? For me it just feels wrong tbh ^^ .. exceptions are "unwanted behaviors", mostly because of data/logic issues, addressed for admins and developers

Reply:
But if it's triggered by a user, and we transmit the exception message to display, then it's important to show it in their preferred language. I understand that it wouldn't be of much help to them, but it still is more welcoming than error messages in a language you don't master.

Reply:
currently we wouldn't display the exception to the user. There's a catch around it and a log-point *gg*. see:
https://github.com/friendica/friendica/blob/0c9163f299e693c853cbcecd05b88523ec9a1a3f/src/Model/Photo.php#L193-L217

But this behavior is now more straight, because before this change, the DB call returned null and the storage call threws an exception. I'll add a generic check at the `Module\Photo` and display the output to the user.

Reply:
This works for me.